### PR TITLE
feat: lang opt on page load

### PR DESCRIPTION
**Pull request by Harshwardhan18**

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,45 +1,70 @@
...
-      if (res) {
-        this.showCitySelection = true;
+  checkLang() {
+    this.paramsSubscription = this.route.queryParams.subscribe((params) => {
+      if (params.lang) {
+        const langReq = params["lang"];
+        if (this.selectedLanguage !== langReq) {
+          confirm(`Are you sure you want to change the language to ${langReq === 'en' ? 'English' : 'Hindi'}`)
+            ? ((this.selectedLanguage = langReq), this.setLanguage())
+            : console.log("good for you");
```
Comment (Manzurkds) [inline src/app/app.component.ts:49]: @Harshwardhan18 Bro, please create a confirmation modal (I think a bootstrap one might already be present there). Never ever use javascript's ugly alert modals

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,45 +1,70 @@
...
-    )
-    .subscribe(res => {
-      if (res) {
-        this.showCitySelection = true;
+  checkLang() {
+    this.paramsSubscription = this.route.queryParams.subscribe((params) => {
+      if (params.lang) {
+        const langReq = params["lang"];
+        if (this.selectedLanguage !== langReq) {
+          confirm(`Are you sure you want to change the language to ${langReq === 'en' ? 'English' : 'Hindi'}`)
```
Comment (Manzurkds) [inline src/app/app.component.ts:47]: @Harshwardhan18 You'll need to maintain a constant as languages array which will have an object that has two keys `id` which will be for example `en` and `value` for example `english`

**Review by Manzurkds — CHANGES_REQUESTED** [review]

**Review by Manzurkds — APPROVED** [review]

File: src/app/app.component.html [hunk 3]
```diff
@@ -1,3 +1,4 @@
+<app-confirmation-modal *ngIf="isConfirmModal" [message]='confirmMessage' (close)='confirmed($event)'></app-confirmation-modal>
```
Comment (avikt) [inline src/app/app.component.html:1]: @Harshwardhan18 Would recommend the boolean to be renamed to something more informative like- `confirmModalOpen`

File: src/app/app.component.ts [hunk 4]
```diff
@@ -1,45 +1,100 @@
...
       if (evt instanceof NavigationEnd) {
-         window.scrollTo(0, 0);
+        window.scrollTo(0, 0);
       }
     });
     this.checkCityPresent();
+    this.checkLang();
   }
 
+  confirmed(event: any) {
```
Comment (avikt) [inline src/app/app.component.ts:58]: @Harshwardhan18 What's this?

File: src/app/app.component.ts [hunk 5]
```diff
@@ -1,45 +1,100 @@
...
+
+  checkLang() {
+    this.paramsSubscription = this.route.queryParams.subscribe((params) => {
+      if (params.lang) {
+        const langReq = params["lang"];
+        this.language = this.languages.find(lang => lang.id === langReq);
+        if (this.selectedLanguage !== langReq) {
+          this.isConfirmModal = true;
+          this.confirmMessage = {
+            title: "Translate",
```
Comment (avikt) [inline src/app/app.component.ts:73]: @Harshwardhan18 Change the title to "Website Language"

File: src/app/app.component.ts (hunk 4 above)
Comment (Harshwardhan18) [reply src/app/app.component.ts:58]: @avikt it responds on how the website will behave upon selection of Ok or cancel for site translation

**Pull request merged**